packages/Slider/src/index.js

```diff
@@ -0,0 +1,148 @@
+import Inferno   from 'inferno'
+import Component from 'inferno-component'
+import { bind }  from 'decko'
+
+export class Slider extends Component {
+  state = {
+    focus: false,
+    keyFocus: false,
+    progress: 40
+  }
+
+  vise = (min, value, max) =>
+    Math.min(Math.max(min, value), max)
+
+  @bind
+  moveSlider({ clientX }) {
+    const { offsetLeft, clientWidth } = this.slider
+    const percentage = (clientX - offsetLeft) / clientWidth
+    const progress = this.vise(0, percentage, 1) * 100
+
+    this.setState({ progress })
+  }
+
+  @bind
+  handleMouseDown(e) {
+    this.setState({ focus: true })
+
+    this.moveSlider(e)
+  }
+
+  @bind
+  handleMouseUp(e) {
+    this.setState({ focus: false })
+  }
+
+  @bind
+  handleMouseMove(e) {
```

Comment:
The **handleMouseMove** event should be applied to the **document** itself when the **handleMouseDown** event is fired so the user will have more freedom to move the slider.